marlowe-test/src/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -243,34 +259,25 @@ instance IsValid Context where
 
 -- | An arbitrary positive integer, mostly small.
 arbitraryPositiveInteger :: Gen Integer
-arbitraryPositiveInteger =
-  frequency
-    [
-      (60, arbitrary `suchThat` (> 0))
-    , (30, arbitraryFibonacci fibonaccis)
-    ]
+arbitraryPositiveInteger = arbitraryNonnegativeInteger <&> \case
+  0 -> 1
+  n -> n
```

Comment:
Rationale: avoid `suchThat` in favour of generating only valid values.

Reply:
That also removes the giving fibonaccis more frequency, not sure whether that is good or bad

Reply:
There isn't anything special about the Fibonaccis, so I don't think it matters